Sources/TuistCache/Cache/URL+Cache.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+import TuistCore
+
+enum CacheURLError: LocalizedError {
+    case missingCloudConfig
+    case incorrectCloudConfig
+}
+
+extension URL {
+    static func apiCacheURL(hash: String, config: Config, contentMD5: String? = nil) throws -> URL {
```

Comment:
This extension is defined to reuse some bits of code from multiple resources, but we ended up with logic that is harder to reason about. *Does the method mean that I contentMD5 is an optional parameter for any endpoint?* I'd rather move the logic back to the individual resources, even if that means we'll be setting in multiple places that the path is `/api/cache` _(not a big deal)_.